src/rax.c

```diff
@@ -523,9 +524,12 @@ int raxGenericInsert(rax *rax, unsigned char *s, size_t len, void *data, void **
      * our key. We have just to reallocate the node and make space for the
      * data pointer. */
     if (i == len && (!h->iscompr || j == 0 /* not in the middle if j is 0 */)) {
+        size_t oldalloc = 0;
+
         debugf("### Insert: node representing key exists\n");
         /* Make space for the value pointer if needed. */
         if (!h->iskey || (h->isnull && overwrite)) {
+            oldalloc = rax_alloc_size(h);
```

Comment:
It looks like we forgot to update `rax->alloc` in the following code.
When `h->iskey == 1 && h->isnull == 1 && overrite == 1`, h will be realloced for data.

```c
        if (h->iskey) {
            if (old) *old = raxGetData(h);
            if (overwrite) raxSetData(h,data);
            errno = 0;
            return 0; /* Element already exists. */
        }
```

Reply:
`rax->alloc` will be updated below on line 553. In the example you provided we return before that line. Yes we should update `rax->alloc` here.